src/java.base/share/classes/java/util/zip/ZipEntry.java

```diff
@@ -702,4 +710,20 @@ public Object clone() {
             throw new InternalError(e);
         }
     }
+
+    /**
+     * Validate that the CEN header size + name length + comment length + extra length
+     *  do not exceed 65,535 bytes per the PKWare APP.NOTE
+     *  4.4.10, 4.4.11, & 4.4.12.
+     * @param name Zip entry name
+     * @param extra Zip extra data
+     * @param comment Zip entry comment
+     * @return true if valid CEN Header size; false otherwise
+     */
+    private static  boolean isCENHeaderValid(String name, byte[] extra, String comment) {
+        int clen = comment == null ? 0 : comment.length();
+        int elen = extra == null ? 0 : extra.length;
+        int headerSize = CENHDR + name.length() + clen + elen;
```

Comment:
`headerSize` may overflow `int`

Reply:
Changed to 
`long headerSize = (long)CENHDR + name.length() + clen + elen;`